src/paper/tasks.py

```diff
@@ -1319,80 +1410,143 @@ def pull_new_openalex_works(page=0, retry=0):
         open_alex = OpenAlex()
 
         while True:
-            # get new works in batches
             works, next_cursor = open_alex.get_new_works_batch(
                 since_date=fetch_since_date,
                 next_cursor=next_cursor
             )
+            # if we've reached the end of the results, exit the loop
             if next_cursor is None or works is None or len(works) == 0:
                 break
 
-            for work in works:
-                _process_openalex_work.apply_async(
-                    (work,),
-                    priority=4,
+            # if we're starting from a specific index, skip until we reach that index
+            if total_papers_processed >= start_index:
```

Comment:
This is fine. I guess there is no way to start at a particular index?

Reply:
Updated it to account of the case where the start_index is inbetween a batch